src/Functions/parseDateTime.cpp

```diff
@@ -1123,6 +1123,14 @@ namespace
                 return cur;
             }
 
+            static Pos mysqlHour12NoLeadingZero(Pos cur, Pos end, const String & fragment, DateTime & date)
```

Comment:
Cosmetic: To stay consistent with the terminology used within the rest of the file: `mysqlHour12SpacePadded`.

Similarly, l. 1142: `mysqlHour24SpacePadded`

Reply:
Cosmetic: Let's reorder l. 1126-1132 after l. 1134-1140. (same for the 24 h variant)

Reply:
Ok, adjusted.

Reply:
BTW why it should have `SpacePadded`, the space padded version is indeed padded with space, I will revert to `WithoutLeadingZero`